### PR TITLE
Refs [TASK][35760] Handle event Home Screen

### PR DESCRIPTION
**Pull request by I-am-Lien**
## Related Tickets
- [#Task 35760: Handle event Home Screen](https://edu-redmine.sun-asterisk.vn/issues/35760)
## WHAT
- Handle event Home Screen
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt [hunk 1]
```diff
@@ -1,19 +1,74 @@
...
+
+    override fun initActions() {
+    }
+
+    private fun updatePieChart(entries: List<PieEntry>?) {
+        val colors = mutableListOf<Int>()
+        context?.resources?.let {
+            colors.apply {
+                add(it.getColor(R.color.color_carnation)!!)
+                add(it.getColor(R.color.color_koromiko)!!)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt]: không assert non-null

File: app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt [hunk 2]
```diff
@@ -1,19 +1,74 @@
...
+        viewModel.ratioData.observe(viewLifecycleOwner, androidx.lifecycle.Observer {
+            updatePieChart(it)
+        })
+    }
+
+    override fun initActions() {
+    }
+
+    private fun updatePieChart(entries: List<PieEntry>?) {
+        val colors = mutableListOf<Int>()
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt:39]: không thay đổi gì thì dùng `listOf` của Color luôn đi em

Comment (I-am-Lien) [reply app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt:39]: có phải thay đổi add color bên dưới đó anh

File: app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt [hunk 3]
```diff
@@ -1,19 +1,74 @@
...
+                add(it.getColor(R.color.color_koromiko))
+            }
+        }
+        val dataSet = PieDataSet(entries, DATA_LABEL)
+        dataSet.colors = colors
+        val data = PieData(dataSet)
+        data.setDrawValues(true)
+        data.setValueFormatter(PercentFormatter(chartRecover))
+        data.setValueTextSize(TEXT_SIZE)
+        data.setValueTextColor(Color.WHITE)
```
Comment (tuannm-1276) [inline app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt]: gom đc ko em nhỉ?

File: app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt [hunk 4]
```diff
@@ -1,19 +1,74 @@
...
+        chartRecover.data = data
+        chartRecover.invalidate()
+    }
+
+    private fun setupPieChart() {
+        chartRecover.isDrawHoleEnabled = true
+        chartRecover.holeRadius = HOLE_RADIUS
+        chartRecover.setTransparentCircleAlpha(TRANSPARENT_ALPHA)
+        chartRecover.description.isEnabled = false
+        chartRecover.animateY(TIME_ANIM, Easing.EaseInOutQuad)
```
Comment (tuannm-1276) [inline app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt]: đây nữa

File: app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt (hunk 3 above)
Comment (I-am-Lien) [reply app/src/main/java/com/sun/excovid19/ui/home/HomeFragment.kt]: được ah

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**